bgpd/bgpd.c

```diff
@@ -6632,15 +6632,8 @@ char *peer_uptime(time_t uptime2, char *buf, size_t len, uint8_t use_json,
 	struct tm *tm;
 
 	/* Check buffer length. */
-	if (len < BGP_UPTIME_LEN) {
-		if (!use_json) {
-			zlog_warn("peer_uptime (): buffer shortage %lu",
-				  (unsigned long)len);
-			/* XXX: should return status instead of buf... */
-			snprintf(buf, len, "<error> ");
-		}
-		return buf;
-	}
+	if (len < BGP_UPTIME_LEN)
+		assert("peer_uptime: buffer is too small" == NULL);
 
```

Comment:
It's a string formatting function... if the buffer isn't long enough, it should just truncate output, like all other string formatting functions do?

Reply:
fixed